### ISSUE TITLE
Lolminer doesn't like my R9 390

### ISSUE THREAD

**Issue by craiga1fix** [Lolliedieb/lolMiner-releases#291]
I've just added an old R9 390 to my system and use Swarm (which calls lolminer) on HiveOS
When benchmarking Ethash, HiveOS showed a gpu crash on amdgpu (error 147 if it matters)

I thought maybe bad riser or card, so fired up in Windows and mined Beam all day with lolminer.
Great!
Switched to Ethash against Nicehash and every share found was reported as a failed share (possible hardware error) in red.

I then pulled the card from riser and ran it onboard. Same issue.
Broken card I thought... So in despair I fired up Phoenix miner and it worked perfectly on Ethash against Nicehash. Successful shares, etc.

Happy to provide whatever logs you need to see what's going on, or remote access to the pc should you need it.

This is with the latest Windows release 1.21 and I believe Swarm is using 1.17 Linux

**Comment by jgonzis** [comment]
Can you try to run the 1.24a if have been lost of improvements, also to r9 390 in last versions... try in Linux the 1.24a and check if works

**Comment by Lolliedieb** [comment]
This might be a memory clock issue ... maybe. 
Can you share the whole setup, clocks and the config string you use (without your btc address for nicehash mining ofc)

**Comment by Kaesarx83** [comment]
I have the same issue on my R290 under Windows 10. I can mine ETC on other miners under Windows but using lolminer I get the same error message. This happens on stock settings or even when I underclock, so this is not a o/c issue.

GPU 0: Detected a defect share (potential hardware error)!

Could I request that we get the newer versions on windows with the zombie mode for r9 290(for eth)?

**Comment by jgonzis** [comment]
> I have the same issue on my R9 290 under Windows 10.
> 
> GPU 0: Detected a defect share (potential hardware error)!
> 
> I can mine ETC on other miners under Windows but using lolminer I get the same error message; this happens on stock settings or even when I underclock, so this is not a o/c issue.
> 
> Lastly, could I request that we get the newer versions on windows with the zombie mode for r9 290(for ETH)?

Hardware error normally is too much OC.

For Windows the Zombie Mode is complicated... You should run it in Windows that will let you to continue mining some more time

**Comment by Kaesarx83** [comment]
> > I have the same issue on my R9 290 under Windows 10.
> > GPU 0: Detected a defect share (potential hardware error)!
> > I can mine ETC on other miners under Windows but using lolminer I get the same error message; this happens on stock settings or even when I underclock, so this is not a o/c issue.
> > Lastly, could I request that we get the newer versions on windows with the zombie mode for r9 290(for ETH)?
> 
> Hardware error normally is too much OC.
> 
> For Windows the Zombie Mode is complicated... You should run it in Windows that will let you to continue mining some more time

I know it is normally too much OC but if you read my actual comment, I am not running overclocks. I actually underclocked the card massively to rule out any core and mem speed issues. 

I have just tried ETC under HiveOS using lolminer and it works fine at the same settings. However, before you say "just use HiveOS" I cannot always be in HiveOS and also I am reporting a bug. The method to fixing a bug is not to "not use it" which I feel is a lot of people's response around here.

Regarding ETH under Windows; I appreciate it isn't easy and that the GPU under Windows has less memory available. However, the DAG size is also quite a bit bigger than the free memory under Linux as well so it's not purely just about the amount of free memory. 

Lolliedieb is using the speed of PCIE (16/8/4?) to swap a partial DAG in and out of GPU from main memory (RAM) with very little slow down. There wasn't even a 0.01 drop in MH/s when the last epoch happened. Therefore, I'm asking Lolliedieb if the same "fix" can be ported to Windows. 

Lastly, it's also not even a case of it being slow under Windows, I cannot not even get ETH to run with lolminer which would also suggest Zombie mode for R9 290 has yet to be implemented in the Windows release at all. I can mine ETH on other miners under Windows albeit very slowly so this is where the PCIE fix would massively help.

Tldr; 
ETC - not overclocked, works fine in Linux with same settings
ETH - doesn't even start in windows, works on other miners (very slowly) - request zombie mode and PCIE fix please

Thanks

**Comment by Lolliedieb** [comment]
R9 290 has zombie mode in Linux, working like a charm there.
Zombie mode in Windows can be considered to be deprecated, because speed is
way to low to be usefull.

Kaesarx83 <notifications@github.com> schrieb am Sa., 27. Feb. 2021, 11:36:

**Comment by Kaesarx83** [comment]
Why would it be useless, if you did the PCIE fix it would be pretty much exactly the same bar a couple of MH/s?

If you don't want to do it then I'd rather you just say but clearly the PCIE fix is working and probably even better than you thought. I'm getting zero slow down on Epoch changes.

**Comment by Kaesarx83** [comment]
Anyway, regardless of the Zombie mode/PCIE Linux fix. (as that's a seperate issue) Lolminer is erroring with hardware errors on R9 within Windows using ETC. 

It's not an OC problem and works fine in other miners on Windows and also works fine on R9 using Linux 1.24. 

Seeing as lolminer is the best and only good ETH miner for R9, I'd rather run lolminer and give you the devfee during the day (ETC) as without lolminer I wouldn't be able to mine ETH at all. Are you able/willing to look at why this is erroring? Let me know if there are any settings I can run to provide you with debug
